lib/quantizeAttributes.js

```diff
@@ -0,0 +1,348 @@
+                                                accessor = primitiveAttributes[attribute];
+                                                if (defined(accessor)) {
+                                                    accessorIds[accessor] = attribute;
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    // Quantize all valid attributes
+    if (quantizeAll) {
+        for (meshId in meshes) {
+            if (meshes.hasOwnProperty(meshId)) {
+                var mesh = meshes[meshId];
+                primitives = mesh.primitives;
+                if (defined(primitives)) {
+                    for (i = 0; i < primitives.length; i++) {
+                        primitive = primitives[i];
+                        attributes = primitive.attributes;
+                        if (defined(attributes)) {
+                            for (attribute in attributes) {
+                                if (attributes.hasOwnProperty(attribute)) {
+                                    if (attribute.indexOf("NORMAL") >= 0 ||
+                                        attribute.indexOf("POSITION") >= 0 ||
+                                        attribute.indexOf("TEXCOORD") >= 0 ||
+                                        attribute.indexOf("JOINT") >= 0 ||
+                                        attribute.indexOf("WEIGHT") >= 0 ||
+                                        attribute.indexOf("COLOR") >= 0
+                                    ) {
+                                        accessor = attributes[attribute];
+                                        accessorIds[accessor] = attribute;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Generate quantized attributes
+    for (var accessorId in accessorIds) {
+        if (accessorIds.hasOwnProperty(accessorId)) {
+            attribute = accessorIds[accessorId];
+            accessor = accessors[accessorId];
+            // as of right now, we only support 32-bit float to 16-bit int quantization
+            if (accessor.componentType != 5126) {
+                break;
+            }
+            accessor.extensions = {
+                "WEB3D_quantized_attributes": {}
+            };
+            attributes = accessor.extensions.WEB3D_quantized_attributes;
+            var accessorData = getAccessorData(gltf, accessorId);
+            var min = accessorData.min;
+            var max = accessorData.max;
+
+            var decodeMatrix = [];
+            var precision = Math.pow(2, 16) - 1;
+            for (j = 0; j < min.length + 1; j++) {
```

Comment:
Is there a more straightforward - even if more verbose - way to create `decodeMatrix`?